### ISSUE TITLE
ZEVA - Reassessment Credit Reduction

### ISSUE THREAD

**Issue by airinggov** [bcgov/zeva#1831]
**Title:** Reassessment Credit Reduction

Blocked by card 1821

**Description:** When the director issues a reassessment (which could have originated as a supplementary report or government reassessment), the original transaction needs to be updated to alter the new credit reduction amounts. This can result in either an increase or decrease of the credits held in the suppliers balance.

Task is to make the credit adjustment to the supplier balance as necessary and to show the new reduction amounts on the credit transaction pages (both bceid and idir versions), see wireframe below, NOTE: this is a redesign of the transaction page that is still in progress (card 1821)

The transactions appear as : YYYY Model Year Report Credit Reduction
The transaction date should not change, it should remain as Sept 30, YYYY 
The transaction ID number should also not change
Just the credit amounts change

**Wireframe:** https://preview.uxpin.com/4946124c44da522837a3b54ccb7b5243fa384039#/pages/163585536/simulate/no-panels?mode=mf

**Purpose and benefit to user:** The vehicle supplier credit balance will be updated accordingly and it will be reflected properly on the credit transaction pages.

**Acceptance Criteria:**

- [ ] Given I am a idir\director, When I am issue a reassessment, then the suppliers credit balance is updated accordingly and it will be reflected properly on the credit transaction pages.

**Development Checklist:**
- [ ] When a reassessment occurs that changes the credit balance, update the credit values of the transaction in the database.
- [ ] Verify nothing else about the aforementioned transaction changes other than the credits.
- [ ] Verify the updated balance is then properly calculated/updated everywhere.

**Notes:**
- When the credit transaction credits values are updated, please check that the supplier overall balance updates. The assumption here is that the supplier balance is calculated by aggregating all transactions.

**Comment by JulianForeman** [comment]
Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Team-Zelda-637d5bc12e6201001055b745/issues/bcgov/zeva/1831?planning-poker) with Zenhub @tim738745

**Comment by kcabhar** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Team-Zelda-637d5bc12e6201001055b745/issues/bcgov/zeva/1831?planning-poker) with Zenhub @AlexZorkin @tim738745

**Comment by jdon-zev** [comment]
Testing this on dev I experienced a problem - see attached screeenshot where the credit reduction is not correctly updated with the reassessment values. ![Ford.jpg](https://images.zenhubusercontent.com/63c843041aa7297691df71de/7e0e0c0b-8d00-4b06-b4bf-8544ded160f5)

**Comment by airinggov** [comment]
Closing as the the scope of this card is completed. There is an issue with the link from the transactions page pointing to the wrong reassessment that may need a new card. There is also an issue with the history component missing the full history and other known issues that still require follow-up.
